marlowe-playground-client/src/Simulation.purs

```diff
@@ -152,32 +152,35 @@ handleAction _ (SetEditorText contents) = do
   editorSetValue contents
   updateContractInState contents
 
-handleAction _ ApplyTransaction = do
+handleAction _ NextSlot = do
+  modifying (_marloweState <<< _Head <<< _slot) (add one)
+  significantSlot <- use (_marloweState <<< _Head <<< to nextSignificantSlot)
+  currentSlot <- use (_marloweState <<< _Head <<< _slot)
   saveInitialState
-  applyTransactions
+  if significantSlot == Just currentSlot then
+    applyInput identity
```

Comment:
Only if continuation evaluates to Close though. It can be `If`, or `Assert`, but not another `When` that hasn't timed out.

Reply:
After an offline discussion with @nau we think this is currently correct